### PR TITLE
Pass TracerListener to constructor of PerfEventVisitor's subclasses, rename PerfEventVisitor::visit to Visit

### PR DESCRIPTION
**Pull request by dpallotti**

### CODE REVIEW

File: src/LinuxTracing/GpuTracepointVisitorTest.cpp [hunk 1]
```diff
@@ -48,12 +48,12 @@ class MockTracerListener : public TracerListener {
 
 class GpuTracepointVisitorTest : public ::testing::Test {
  protected:
-  void SetUp() override { visitor_.SetListener(&mock_listener_); }
+  void SetUp() override {}
 
   void TearDown() override {}
```
Comment (beckerhe) [inline src/LinuxTracing/GpuTracepointVisitorTest.cpp:53]: Nit: I think you could just remove those. They aren't pure virtual and using constructors/destructors over SetUp/TearDown when possible is encouraged as well (http://google.github.io/googletest/faq.html#CtorVsSetUp)

File: src/LinuxTracing/LostAndDiscardedEventVisitorTest.cpp [hunk 2]
```diff
@@ -63,21 +63,18 @@ class MockTracerListener : public TracerListener {
 
 class LostAndDiscardedEventVisitorTest : public ::testing::Test {
  protected:
-  void SetUp() override { visitor_.SetListener(&mock_listener_); }
+  void SetUp() override {}
 
   void TearDown() override {}
```
Comment (beckerhe) [inline src/LinuxTracing/LostAndDiscardedEventVisitorTest.cpp:68]: Nit: As above

**Review by beckerhe — APPROVED** [review]
Thanks!

File: src/LinuxTracing/GpuTracepointVisitorTest.cpp (hunk 1 above)
Comment (dpallotti) [reply src/LinuxTracing/GpuTracepointVisitorTest.cpp]: Yep, done.

File: src/LinuxTracing/LostAndDiscardedEventVisitorTest.cpp (hunk 2 above)
Comment (dpallotti) [reply src/LinuxTracing/LostAndDiscardedEventVisitorTest.cpp]: Done.

**Pull request merged**